### PR TITLE
add error_stack on reject also reject error not string

### PR DESCRIPTION
**Pull request by dreadjr**
Wanted to start a discussion about 2 things,

1. adding error_stack
1. reject new Error() vs string

### CODE REVIEW

**Comment by dreadjr** [comment]
I signed it!

**Comment by drtriumph** [comment]
I'll take a look at this today - initially I used strings because they are easiest to represent in Firebase, and perhaps we don't want to coerce every rejection with a string into an Error object, but a stack trace would be a useful addition

**Comment by dreadjr** [comment]
Ok cool, yeah i wasn't sure about the error stack as far as privacy/security is concerned either. It might need to go in an "errors" bucket and just reference it. I was thinking could just do a `JSON.stringify` as well.

**Comment by drtriumph** [comment]
In using the queue, I've tended to lock down the `/tasks` location to write only (and create only) for clients that push items onto the queue, and only allowed the trusted server processes read access. This means that for me, adding the stack trace to the task is a non-issue.

I'm curious whether that's a pattern other people are using, or whether read access is indeed required in some use-cases? If read access is required, then you're right that adding stack traces could be a security concern

File: src/lib/queue_worker.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ QueueWorker.prototype._resetTask = function(taskRef, deferred) {
       } else {
         var errorMsg = 'reset task errored too many times, no longer retrying';
         logger.debug(self._getLogEntry(errorMsg), error);
-        deferred.reject(errorMsg);
+        deferred.reject(new Error(errorMsg));
```
Comment (drtriumph) [inline src/lib/queue_worker.js]: I'm not sure how useful this being an error would be - won't the stack trace always be an internal Firebase Queue node module trace? Won't they be able to get that information from the error message string itself?

Comment (drtriumph) [inline src/lib/queue_worker.js]: Ah, sorry, was confused for a second between `reject` and `_reject` - this could be useful

**Comment by dreadjr** [comment]
Yeah true. I was thinking too of converting stack trace into Jain so it could be saved to firebase more easily.

File: src/lib/queue_worker.js (hunk 1 above)
Comment (dreadjr) [inline src/lib/queue_worker.js]: Event though the stack trace might not be helpful in all cases, it's nice just to be consistent. Also might help later on if more of the internal api is exposed and can be built upon.

**Comment by drtriumph** [comment]
I included these changes in #24 with a few minor adjustments. The major changes were
1. We don't specifically coerce all errors to Error objects - if they're strings then they behave the same as they do now (without a stack trace)
2. There's now a `suppressStack` option for the queue that prevents stack traces from ending up in the Firebase

Take a look and let me know if there's something I missed. Let's move the conversation there

**Pull request closed without merging**